src/core/Interpreter.ts

```diff
@@ -147,13 +143,23 @@ class Interpreter {
             // TODO the description and extras should be different for reverted transactions
         }
 
-        const interpretationMapping = (toAddress && this.contractSpecificInterpreters[toAddress]) || null
-        const methodSpecificMapping = (methodName && interpretationMapping?.writeFunctions[methodName]) || null
+        const interpretationMapping: InterpreterMap | null = (toAddress && this.contractSpecificInterpreters[toAddress.toLowerCase()]) || null
+        let methodSpecificMappings: MethodMap[] = [];
+
+        if (multicallFunctionNames.includes(methodName || '') && multicallContractAddresses.includes(toAddress || '')) {
+            traceCalls?.forEach((call) => {
+                if (interpretationMapping?.writeFunctions[call.name || '']) {
```

Comment:
you could name `interpretationMapping?.writeFunctions[call.name || '']` something that explains what it is, and then use that var in the two places it's used for a small improvement in readability